client/components/SettingsPane.tsx

```diff
@@ -0,0 +1,190 @@
+        // convert from ms to seconds
+        setWindowSize(Number(e.target.value) / 1000);
+    };
+
+    const onRateLimiterChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+        setRateLimiterType(e.target.value as RateLimiterType);
+    };
+
+    const onUpdate = (e: React.FormEvent<HTMLButtonElement>, saveSettings = false) => {
+        e.preventDefault();
+        // TODO: Window size math
+        const updatedConfig: RateLimiterConfig = {
+            type: rateLimiterType,
+            options: {
+                capacity,
+                ...((rateLimiterType as WindowType) && { windowSize: windowSize * 1000 }),
+                ...((rateLimiterType as BucketType) && { refillRate }),
+            },
+        };
+        setRateLimiterConfig(updatedConfig, saveSettings);
+    };
+
+    return (
+        <>
+            <h3>Settings</h3>
+            {rateLimiterLoading ? (
+                <div className="loader" />
+            ) : (
+                <div className="panelColumn">
+                    <select
+                        className="panelDropdown"
+                        value={rateLimiterType}
+                        onChange={onRateLimiterChange}
+                    >
+                        <option disabled value="None">
+                            Select an Option
+                        </option>
+                        <option value="FIXED_WINDOW">Fixed Window</option>
+                        <option value="LEAKY_BUCKET">Leaky Bucket</option>
+                        <option value="TOKEN_BUCKET">Token Bucket</option>
+                        <option value="SLIDING_WINDOW_LOG">Sliding Window Log</option>
+                        <option value="SLIDING_WINDOW_COUNTER">Sliding Window Counter</option>
+                    </select>
+                    <Slider
+                        name="Capacity"
+                        value={capacity}
+                        displayValue={capacity}
+                        onChange={onCapacityChange}
+                        min={0}
+                        max={500}
+                        unit="tokens"
+                    />
+                    {['TOKEN_BUCKET', 'LEAKY_BUCKET'].includes(rateLimiterType) ? (
+                        <Slider
+                            name="Rate"
+                            value={refillRate}
+                            displayValue={refillRate}
+                            onChange={onRefillRateChange}
+                            min={0}
+                            max={100}
+                            unit="tokens/s"
+                        />
+                    ) : (
+                        <Slider
+                            name="Window Size"
+                            value={windowSize * 1000} // Slider goes down to millisecond for granularity
+                            displayValue={windowSize} // Slider goes down to millisecond for granularity
```

Comment:
adjusting to the millisecond over 10 minutes seems to granular to me

Reply:
Changed this to 1 minutes